### ISSUE TITLE
About the best learning rate you used for the vanilla Adafactor across different model sizes of LLaMA and GPT-2 in your experiments.

### ISSUE THREAD

**Issue by jie040109** [zyushun/Adam-mini#39]
Dear authors,
I’m very curious about the best learning rate you used for the vanilla Adafactor across different model sizes of LLaMA and GPT-2 in your experiments, as this detail doesn’t seem to be clearly specified in the paper.

**Comment by zyushun** [comment]
Hi @jie040109 , thanks for the question. Here is the learning rate that we use for Adafactor.

1. In Figure 8 (a) (GPT2-125M): we use lr = 6e-4. We chose this lr because the recommended Adam lr in this setting is 6e-4 (according to the grid search by https://arxiv.org/abs/2305.14342). We use lr = 6e-4 for all methods here. 
2. In Figure 13 (a)  (Llama 2-20M): we sweep the lr for Adafactor and Adafactor-zhai-version while keeping other hyperparms as default. All lr performs similarly for Adafactor. The optimal lr = 5e-3 for Adafactor-Zhai-version. We find both versions of Adafactor perform worse than Adam-mini (lr = 5e-3).
3.  In Figure 13 (b)  (Llama 2-1B): we sweep the lr for Adafactor and Adafactor-zhai-version while keeping other hyperparms as default.  All lr performs similarly for Adafactor. All lr fails for Adafactor-Zhai-version.  Both versions of Adafactor perform worse than Adam-mini (lr = 6e-4).
4. We sweep over other hyperparameters of Adafactor as well. See  Results in Appendix D.7.

**Comment by jie040109** [comment]
Thanks for your prompt reply, but it seems that you only ran a short number of rounds (20B tokens) on gpt2 124M. I found that Adafactor would crash when using the learning rate you mentioned when training for a longer number of rounds(10w steps). Here, AdamW uses all the hyperparameters recommended by nanogpt and trains for 10w steps.
<img width="775" alt="Image" src="https://github.com/user-attachments/assets/80ba04ee-c752-4b66-a358-dd3695045a16" />
